brontide/noise.go

```diff
@@ -743,7 +748,19 @@ func (b *Machine) WriteMessage(w io.Writer, p []byte) error {
 // ReadMessage attempts to read the next message from the passed io.Reader. In
 // the case of an authentication error, a non-nil error is returned.
 func (b *Machine) ReadMessage(r io.Reader) ([]byte, error) {
-	_, err := io.ReadFull(r, b.nextCipherHeader[:])
+	conn, ok := r.(net.Conn)
+	if !ok {
+		return nil, fmt.Errorf("net.Conn required to read message")
+	}
+
+	// Set the read timeout to zero, which will cause this method to block
```

Comment:
I'm not sure we actually want this. If the other party writes half the header, then we'll just sit here. Since this'll block forever, the idle timer in the peer also won't be able to tick on its own and disconnect the peer.

Reply:
that is the current behavior

Reply:
also the idle timer fires a `Disconnect` via an `AfterFunc`, so blocking arbitrarily long doesn't prevent the timer from being triggered: https://github.com/lightningnetwork/lnd/blob/master/peer.go#L958